src/Linters/ShoutCaseEnumMembersLinter.hack

```diff
@@ -0,0 +1,129 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+use namespace HH\Lib\{C, Dict, Regex, Str, Vec};
+
+final class ShoutCaseEnumMembersLinter extends AutoFixingASTLinter {
+  const type TContext = EnumDeclaration;
+  const type TNode = Enumerator;
+
+  /* @var enum_name -> shout_name to original_name*/
+  private static dict<string, dict<string, string>> $newNames = dict[];
+
+  <<__Override>>
+  public function getLintErrorForNode(
+    self::TContext $enum,
+    self::TNode $member,
+  ): ?ASTLintError {
+    $original_name = self::memberToName($member);
+    if (self::isShoutCase($original_name)) {
+      return null;
+    }
+    self::prepareAllNewNamesFor($enum);
+    $enum_name = $enum->getName()->getText();
+    $shout_name = self::transformToShoutCase($original_name);
+
+    // If we can't find the shout name, we bailed, because the enum
+    // member names were not unique.
+    // If the shout name maps to something other than the orignal,
+    // we would cause a collision.
+    // We bail out either way.
+    if (
+      !C\contains_key(self::$newNames[$enum_name], $shout_name) ||
+      self::$newNames[$enum_name][$shout_name] !== $original_name
+    ) {
+      return new ASTLintError(
+        $this,
+        Str\format(
+          'Member {%s} is not in SHOUT_CASE,
+but the autogenerated name collides or your enum contains duplicate names,
+so no autofix is suggested.',
+          $original_name,
+        ),
+        $member,
+        null,
+      );
+    }
+
+    return new ASTLintError(
+      $this,
+      Str\format('Member {%s} is not in SHOUT_CASE', $original_name),
+      $member,
+      () ==> self::getFixedNode($member),
+    );
+  }
+
+  <<__Memoize>>
+  private static function isShoutCase(string $member_name): bool {
+    return Regex\matches($member_name, re"/^[A-Z0-9_]+$/");
+  }
+
+  private static function memberToName(Enumerator $member): string {
+    return $member->getName()->getLastTokenx()->getText();
+  }
+
+  private static function prepareAllNewNamesFor(EnumDeclaration $enum): void {
+    $enum_name = $enum->getName()->getText();
+    if (C\contains_key(self::$newNames, $enum_name)) {
+      return;
+    }
+
+    $old_names = Vec\map(
+      $enum->getChildren()['enumerators']->toVec(),
+      $member ==> self::memberToName($member as Enumerator),
+    );
+    $old_names_unique = keyset($old_names);
+
+    // !C\all_unique() -> bail
+    if (C\count($old_names_unique) !== C\count($old_names)) {
```

Comment:
Is this check needed? How could the old names not be unique?

Reply:
Will hhast refuse to parse enums with duplicates in them from the start? I was trying to protect against that, but I didn't actually test that. Will do tomorrow.

Reply:
> Will hhast refuse to parse enums with duplicates in them from the start?

I'm not sure but it's definitely a Hack error. I think it's fine for linters to have undefined behavior when dealing with invalid code -- probably better than adding more code to handle various invalid cases, and making the linter more complicated.